stratum/hal/lib/barefoot/bf_chassis_manager.cc

```diff
@@ -578,31 +653,34 @@ ::util::Status BFChassisManager::ReplayPortsConfig(uint64 node_id) {
           << "fec_mode field should contain a value";
     }
 
-    RETURN_IF_ERROR(bf_pal_interface_->PortAdd(unit, port_id, *config.speed_bps,
+    ASSIGN_OR_RETURN(auto sdk_port_id, GetSdkPortId(node_id, port_id));
```

Comment:
```
stratum/hal/lib/barefoot/bf_chassis_manager.cc:656:40: warning: calling function 'GetSdkPortId' requires holding mutex 'chassis_lock' [-Wthread-safety-analysis]
    ASSIGN_OR_RETURN(auto sdk_port_id, GetSdkPortId(node_id, port_id));
```

Reply:
good catch! clang ftw. moved call out of lambda